internal/deephash/deephash.go

```diff
@@ -88,6 +89,20 @@ func print(w *bufio.Writer, v reflect.Value, visited map[uintptr]bool) (acyclic
 				w.Write(b)
 				return true
 			}
+		case netaddrIPPort:
+			var b []byte
+			var err error
+			if v.CanAddr() {
+				x := v.Addr().Interface().(*netaddr.IPPort)
+				b, err = x.MarshalText()
```

Comment:
How about calling As16() and Zone() string and writing those, instead of allocating a []byte here?

Reply:
Combining https://github.com/inetaf/netaddr/pull/168 and a `*[]byte` will make this zero-allocs. I have it running locally happily, just tinkering around the edges before I push it up.

Reply:
Also, I think we could replace the sync.Pools in this package with a local variable tied to the lifetime of a particular Hash call, if we wanted. Maybe lower priority.

Reply:
Also, aren't you on vacation? This can wait until you get back. I'll just keep tinkering until then. :)